portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +24,55 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    
+    // instance vairables 
+    ArrayList<String> messages;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("Hello Fran you did it you solved!!");
+
+    // only show the messages and print json if messages not null
+    if(messages != null){
+      // converting the message array list to a json 
+        String jsonString = convertToJsonUsingGson(messages);
+
+        // send the json as the response  
+        response.setContentType("application/json;");
+        response.getWriter().println(jsonString);
+    }
+
+  }
+
+   @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // recieve the comments from response.getParamaters 
+    String userName = getParamatersDataMassaged(request, "username");
+    String specificComment = getParamatersDataMassaged(request, "comment");
+
+    // TODO: make an object for the comments to take all the info (name, time, video/picture)
+    String completeComment = userName + " says " + specificComment; 
+    
+    // create new message array if null
+    if(messages == null){
+       messages = new ArrayList<String>();
+    }
+    //add the comments to the messages array
+    messages.add(completeComment);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
+  // method that takes in an arraylist of string and returns in json string form 
+  private String convertToJsonUsingGson(ArrayList<String> messagesList) {
+    Gson gson = new Gson();
+    String json = gson.toJson(messagesList);
+    return json;
+  }
+
+   /**  returns the massaged version of getting the data from the request*/
+  private String getParamatersDataMassaged(HttpServletRequest request, String parameterName) {
+    return request.getParameter(parameterName);
```

Comment:
What happens here if the user doesn't fill out this part of the form?

Reply:
Changed this method to return a boolean representing if the userInput is valid. Currently valid means non null